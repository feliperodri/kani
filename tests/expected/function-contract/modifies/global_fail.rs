// Copyright Kani Contributors
// SPDX-License-Identifier: Apache-2.0 OR MIT
// kani-flags: -Zfunction-contracts

static mut PTR: u32 = 0;

#[kani::requires(PTR < 100)]
unsafe fn modify() {
    PTR += 1;
}

#[kani::proof_for_contract(modify)]
fn main() {
    unsafe {
        modify();
    }
}
