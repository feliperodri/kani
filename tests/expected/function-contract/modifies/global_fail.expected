assigns\
- Status: FAILURE\
- Description: "Check that *var_1 is assignable"\
in function modify_wrapper

Failed Checks: Check that *var_1 is assignable\
in modify_wrapper

VERIFICATION:- FAILED
